include/wx/localedefs.h

```diff
@@ -108,6 +108,83 @@ struct WXDLLIMPEXP_BASE wxLanguageInfo
     const char* TrySetLocale() const;
 };
 
+// ----------------------------------------------------------------------------
+// wxLocaleIdent: allows to fully identify a locale under all platforms
+// ----------------------------------------------------------------------------
+
+class WXDLLIMPEXP_BASE wxLocaleIdent
+{
+public:
+    // Leave language empty
+    wxLocaleIdent() { }
+
+    // Construct from language, i.e. a two-letter ISO 639-1 code (or a
+    // three-letter ISO 639-2 code if there is no ISO 639-1 code for this
+    // language).
+    wxLocaleIdent(const char* language)
+        : m_language(wxString::FromAscii(language))
+    {
+    }
+
+    // Set language
+    wxLocaleIdent& Language(const wxString& language)
+    {
+        m_language = language;
+        return *this;
+    }
+
+    // Set region
+    wxLocaleIdent& Region(const wxString& region)
+    {
+        m_region = region;
+        return *this;
+    }
+
+    // Set script (not supported under Unix)
+    wxLocaleIdent& Script(const wxString& script)
+    {
+        m_script = script;
+        return *this;
+    }
+
+    // Set charset (only supported under Unix)
+    wxLocaleIdent& Charset(const wxString& charset)
+    {
+        m_charset = charset;
+        return *this;
+    }
+
+    // Set modifier (only supported under Unix)
+    wxLocaleIdent& Modifier(const wxString& modifier)
+    {
+        m_modifier = modifier;
+        return *this;
+    }
+
+    // Accessors for the individual fields.
+    const wxString& GetLanguage() const { return m_language; }
+    const wxString& GetRegion() const { return m_region; }
+    const wxString& GetScript() const { return m_script; }
+    const wxString& GetCharset() const { return m_charset; }
+    const wxString& GetModifier() const { return m_modifier; }
+
+    // Construct platform dependent name
+    wxString GetName() const;
```

Comment:
This can return an empty string; again the meaning of that is unclear.

Reply:
I was thinking about renaming this to something like `GetPlatformName()` and will probably do it.

Also, this method is mostly useful for the implementation, i.e. used in the platform-specific code, but I thought we could expose it as long as we had it anyhow. Do you think we shouldn't?

Reply:
No, having a way to get platform name is certainly useful.